### PR TITLE
[mesonbuild] update meson build

### PR DESCRIPTION
**Pull request by lhs8928**
- Make yocto build option
 - Check sysroot

Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -25,6 +26,13 @@ if get_option('enable-tizen')
     add_project_arguments('-D__FEATURE_CHECK_SUPPORT__', language: ['c', 'cpp'])
   endif
 endif
+if get_option('enable-yocto')
+  if build_platform == ''
+    build_platform = 'yocto'
+  else
+    error(multi_build_platform_err_msg_template.format(build_platform, 'yocto'))
```
Comment (jijoongmoon) [inline meson.build:33]: This is little bit strange. is there no case in which the build_paltform='yocto' ?? If it is, then there will be error.

**Review by jijoongmoon — APPROVED** [review]
LGTM except for small comments.

File: meson.build (hunk 1 above)
Comment (myungjoo) [reply meson.build:33]: Logic of 29~35 is overly complex and is not general (extensible). Simplify it.

```
count = 0
if TIZEN
   ...
   count ++
endif
if YOCTO
   ...
   count ++;
endif

if (count > 1) error
```
should be enough.

File: meson_options.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
 option('enable-tizen', type: 'boolean', value: false)
+option('enable-yocto', type: 'boolean', value: false)
```
Comment (myungjoo) [inline meson_options.txt:2]: Refer to https://mesonbuild.com/Build-options.html
```
option('platform', type: 'combo', choices: ['none', 'tizen', 'yocto'], value: 'none')
```
might be more pretty.

**Review by myungjoo — APPROVED** [review]
LGTM.

**Pull request merged**